frontend/src/packages/auth/auth-api.ts

```diff
@@ -35,6 +36,18 @@ class AuthApi extends BaseHttpApi {
 
     return await response.json<UserSignUpResponseDto>();
   }
+
+  public async getAuthenticatedUser(): Promise<UserAuthResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(AuthApiPath.USER, {}),
+      {
+        method: 'GET',
+        hasAuth: true,
+      },
+    );
+
+    return await response.json<UserAuthResponseDto>();
```

Comment:
```suggestion
    return response.json<UserAuthResponseDto>();
```
The async function will wrap with Promise